R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,214 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
+
+# Create a control table where year 2010 is the control period and renamed 1975, discarding all other years.
+     L251.ssp2_ef %>%
+      select(-emiss.coeff) %>%
+      subset(year == min(year)) %>%
+      mutate(year = 1975,
+             ctrl.name ="GDP_control") ->
+      L251.ctrl.delete
+
+# This section adds emissions controls for future years of vintaged technologies for SSP emission factors.
+     L251.ssp15_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp15_ef_vin
+
+     L251.ssp2_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp2_ef_vin
+
+     L251.ssp34_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp34_ef_vin
+
+# This section renames SO2 variables so that it has regional SO2 emission species.
+     L251.ctrl.delete <- rename_SO2( L251.ctrl.delete, A_regions, FALSE )
+     L251.ssp15_ef <- rename_SO2( L251.ssp15_ef, A_regions, FALSE )
+     L251.ssp2_ef <- rename_SO2( L251.ssp2_ef, A_regions, FALSE )
+     L251.ssp34_ef <- rename_SO2( L251.ssp34_ef, A_regions, FALSE )
+     L251.ssp15_ef_vin <- rename_SO2( L251.ssp15_ef_vin, A_regions, FALSE )
+     L251.ssp2_ef_vin <- rename_SO2( L251.ssp2_ef_vin, A_regions, FALSE )
+     L251.ssp34_ef_vin <- rename_SO2( L251.ssp34_ef_vin, A_regions, FALSE )
+
+# This section deletes GDP control functions that exist for the supply sector in all
+# regions for non-CO2 emission species for the 1975 control period.
```

Comment:
This comment is really hard to follow (the prepositional phrases appear to be nested four deep).  Could you reword it for clarity?  Also, a note as to _why_ we need to do this would be helpful.